### PR TITLE
Bump backpack to 4.0

### LINKED ISSUE #1514: Task - Admin - Update Laravel Backpack to 4.x
# Description
Laravel Backpack has a shiny new 4.0 update that should be merged in soon. The codebase needs a bit of work in order to make it happen.

[Check out the changelog](https://github.com/Laravel-Backpack/CRUD/blob/4.0.0/CHANGELOG.md#removed) for more info on what needs changing.

## Note:
Depends on #1301 being completed first and merged in

### PR DESCRIPTION
**Pull request by Khristophor**
Resolves #1514

### Notes:
- Requires #1301

### CODE REVIEW

File: app/Http/Controllers/Admin/ManagerCrudController.php [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 class ManagerCrudController extends CrudController
```
Comment (tristan-orourke) [inline app/Http/Controllers/Admin/ManagerCrudController.php]: The search function on this page doesn't seem to work. To be fair, I think this is an old issue, so I don't know if you want to fix it in this PR.

Comment (Khristophor) [reply app/Http/Controllers/Admin/ManagerCrudController.php]: I can take a quick look

File: app/Http/Controllers/Admin/JobPosterCrudController.php [hunk 2]
```diff
@@ -71,7 +75,10 @@ function ($entry) {
                         $entry->submitted_applications_count();
                 }
         ]);
+    }
 
+    public function setupUpdateOperation()
+    {
```
[the remark this answers is not in the document]
Comment (Khristophor) [reply app/Http/Controllers/Admin/JobPosterCrudController.php]: I think these are resolved, not sure what the difference is now but the form submits these values as `null` if the checkbox is unchecked. I added a couple mutators to massage the fields on submission.

File: app/Http/Controllers/Admin/SkillCrudController.php [hunk 3]
```diff
@@ -37,6 +34,80 @@ public function setup() : void
...
+                'name' => 'classifications',
+                'key' => 'classifications_filter',
+                'type' => 'select2_multiple',
+                'label' => 'Filter by classification'
+            ], function () {
+                // The options that show up in the select2.
+                return Classification::all()->pluck('key', 'id')->toArray();
+            }, function ($values) {
+                // If the filter is active.
+                foreach (json_decode($values) as $key => $value) {
```
[the remark this answers is not in the document]
Comment (tristan-orourke) [reply app/Http/Controllers/Admin/SkillCrudController.php]: I tried looking into this. Straight-up removing lines 90-96 of SkillCrudController (the third argument of addFilter) seemed to fix it - though I'm not sure why it was there, so I could be missing something.

[the remark this answers is not in the document]
Comment (Khristophor) [reply app/Http/Controllers/Admin/SkillCrudController.php]: Hmm not sure. In each instance it seemed to be non-nullable checkbox fields being handed back to the server as `nulls` so I just added some mutators to Job Poster, Skill and User.

[the remark this answers is not in the document]
Comment (tristan-orourke) [reply app/Http/Controllers/Admin/SkillCrudController.php]: This error is different: `json_decode expects parameter 1 to be a string, array given`. It happens when you add classifications to the skill and try to submit.

**Pull request merged**